src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.NameGenerator.cs

```diff
@@ -13,15 +15,40 @@ internal partial class DeclarationNameCompletionProvider
     {
         internal class NameGenerator
         {
-            internal static ImmutableArray<Words> GetBaseNames(ITypeSymbol type)
+            internal static ImmutableArray<Words> GetBaseNames(ITypeSymbol type, bool pluralize)
             {
                 var baseName = TryRemoveInterfacePrefix(type);
                 var parts = StringBreaker.GetWordParts(baseName);
                 var result = GetInterleavedPatterns(parts, baseName);
+                if (pluralize)
+                {
+                    result = Pluralize(result);
+                }
+
                 parts.Free();
                 return result;
             }
 
+            private static ImmutableArray<Words> Pluralize(ImmutableArray<Words> baseNames)
+            {
+                var result = ArrayBuilder<Words>.GetInstance();
+                foreach (var baseName in baseNames)
+                {
+                    var lastWord = baseName[baseName.Length - 1];
```

Comment:
Looks like it can do that :)   https://github.com/jeremydw/google-apputils-python/blob/master/tests/humanize_test.py#L37